shared/src/store/action-reducers/vehicle.ts

```diff
@@ -228,15 +240,57 @@ export namespace VehicleActionReducers {
                     x,
                     y: unloadPosition.y,
                 };
+                SpatialTree.addElement(
+                    draftState,
+                    'personnel',
+                    person.id,
+                    person.position
+                );
             }
-            for (const currentMaterial of materials) {
+
+            for (const material of materials) {
                 x += space;
-                currentMaterial.position ??= {
+                material.position ??= {
                     x,
                     y: unloadPosition.y,
                 };
+                SpatialTree.addElement(
+                    draftState,
+                    'materials',
+                    material.id,
+                    material.position
+                );
             }
-            calculateTreatments(draftState);
+
+            /**
+             * Has the Ids of every personnel and material that we calculated already,
+             * so we don't recalculated multiple times.
+             * Without this set of skipped UUIDs for each material and personnel
+             * around each patient that was unloaded
+             * calculateTreatments would probably calculate the them multiple times
```

Comment:
Please look into the grammar, etc. of this comment. I think I know what you mean, but it's not very clear.